dev/src/DocFx/Node/ClassNode.php

```diff
@@ -0,0 +1,170 @@
+<?php
+/**
+ * Copyright 2022 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Dev\DocFx\Node;
+
+use SimpleXMLElement;
+
+class ClassNode
+{
+    use DocblockTrait;
+    use NameTrait;
+
+    private $childNode;
+
+    public function __construct(
+        private SimpleXMLElement $xmlNode
+    ) {}
+
+    public function isProtobufEnumClass(): bool
+    {
+        if ($this->getExtends()) {
+            return false;
+        }
+
+        if (empty($this->xmlNode->docblock)) {
+            return false;
+        }
+
+        if (empty($this->xmlNode->docblock->{'long-description'})) {
+            if (empty($this->xmlNode->docblock->{'description'})) {
+                return false;
+            }
+            $description = $this->xmlNode->docblock->{'description'};
+        } else {
+            $description = (string) $this->xmlNode->docblock->{'long-description'};
+        }
+
+        // check that last line of long-description starts with "Protobuf type..."
+        $descriptionParts = explode("\n", $description);
+        $lastDescriptionLine = array_pop($descriptionParts);
+        return 0 === strpos($lastDescriptionLine, 'Protobuf type');
+    }
+
+    public function isGapicEnumClass(): bool
+    {
+        // returns true if the class extends \Google\Protobuf\Internal\Message
+        return false !== strpos($this->getNamespace(), '\Enums\\');
+    }
+
+    public function isServiceClass(): bool
+    {
+        // returns true if the class extends a generated GAPIC client
+        if ($extends = $this->getExtends()) {
+            return 'GapicClient' === substr($extends, -11);
+        }
+        return false;
+    }
+
+    public function getStatus(): string
+    {
+        if (!$this->xmlNode->docblock) {
+            return '';
+        }
+
+        foreach ($this->xmlNode->docblock->tag as $tag) {
+            if ((string) $tag['name'] === 'deprecated') {
+                return 'deprecated';
+            }
+        }
+
+        return '';
+    }
+
+    public function getExtends(): string
+    {
+        return (string) $this->xmlNode->extends;
+    }
+
+    public function isInternal(): bool
```

Comment:
This method is returning strings `''` and `'true'`, do we need to update the return type or modify the return values?

Reply:
good catch.. this must have been carried over from when this was YAML output, and thanks to `strict_types` not being used, this was casting the string values to a boolean.

I've fixed it!